src/modules/api/user/user.controller.ts

```diff
@@ -1,7 +1,43 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
 import { UserService } from './user.service';
+import { IResult } from 'src/common/interfaces/response';
+import { UserDTO } from './user.dto';
 
-@Controller('users/:id')
+@Controller('users')
 export class UserController {
-  constructor(private userService: UserService) {}
+  constructor(private readonly userService: UserService) {}
+
+  @Get(':id')
+  async getUserById(@Param('id') id) {
+    try {
+      const user = await this.userService.getUserById(id);
+      if (user) {
+        const userResp: UserDTO = {
+          name: user.name,
+          _id: user._id,
+          emailAddress: user.emailAddress,
+          joinedAt: user.joinedAt,
+          phoneNumber: user.phoneNumber,
+        };
+        const resp: IResult<UserDTO> = {
+          status: 'OK',
+          result: userResp,
+        };
+        return resp;
+      }
+      const errResp: IResult<UserDTO> = {
+        status: 'NOT OK',
+        error: 'User not found',
+        result: null,
```

Comment:
make result as optional in IResult just like error

Reply:
I dont think we should do that..if result is null, that indicates that we should be displaying the error message right ? The response must always have a result key i guess.

Reply:
```
var obj = { 
    status : 'NOT OK',
    error : 'Error occurred'
}

// obj['result'] will give us null here
```
and dont make changes to the IResult interface, the latest master has the changes. Just remove `result : null`